Model/Behavior/UploadBehavior.php

```diff
@@ -1508,12 +1508,12 @@ protected function _createImageResource($filename, $pathInfo) {
  */
 	protected function _imagecreatefromjpegexif($filename) {
 		$image = imagecreatefromjpeg($filename);
-		$exif = false;
+		$exif = array();
 		if (function_exists('exif_read_data')) {
 			$exif = exif_read_data($filename);
 		}
 
-		if ($image && $exif && isset($exif['Orientation'])) {
+		if ($image && isset($exif['Orientation']) == true) {
```

Comment:
`exif_read_data` can return `false`, so assuming it is an array isn't a great idea.